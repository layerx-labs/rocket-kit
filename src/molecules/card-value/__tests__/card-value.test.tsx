import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { CardValue } from '../../..';

describe('Card Value', () => {
  it('renders', () => {
    const { asFragment } = render(
      <CardValue label={'Amount'} value={'1234567'} currency="tkai" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct label and string value', () => {
    const label = 'Accounts';
    const value = '345';
    render(<CardValue label={label} value={value} currency="tkai" />);
    screen.getByText(label);
    screen.getByText(value);
  });

  it('has correct label and number value', () => {
    const label = 'Accounts';
    const value = 345;
    render(<CardValue label={label} value={value} currency="tkai" />);
    screen.getByText(label);
    screen.getByText(value);
  });

  it('does not show arrow button', () => {
    render(<CardValue label={'Amount'} value={'1234567'} currency="tkai" />);
    expect(screen.queryByRole('button')).not.toBeInTheDocument();
  });

  it('calls action callback', () => {
    const description =
      'Asymmetrical tbh irony echo park four dollar toast chia';
    const onClickAction = jest.fn();
    render(
      <CardValue
        label={'Amount'}
        value={'1234567'}
        currency="tkai"
        description={description}
        buttonValue="Open"
        onClick={onClickAction}
      />
    );

    userEvent.click(screen.getByRole('button'));
    screen.getByText(description);
    expect(onClickAction).toBeCalledTimes(1);
  });
});
