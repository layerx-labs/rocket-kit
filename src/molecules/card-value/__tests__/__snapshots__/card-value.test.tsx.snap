// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Value renders 1`] = `
<DocumentFragment>
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.75rem;
  font-weight: 700;
  color: #939393;
  text-transform: uppercase;
}

.c1 svg {
  width: auto;
  height: 0.9375rem;
}

.c0 {
  border: 1px solid #f4f4f4;
  border-radius: 6px;
  background-color: #ffffff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.9375rem;
  overflow-x: auto;
}

.c0 > span {
  display: block;
  margin-top: 0.3125rem;
  font-size: 2rem;
  font-weight: var(--bold,700);
  line-height: 1;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

@media (min-width:1280px) {
  .c0 {
    padding: 1.875rem;
  }
}

<div
    class="c0 card-value"
  >
    <label
      class="c1 label"
    >
      Amount
      <svg
        class="icon"
        viewBox="0 0 32 32"
      >
        <path
          d="M9.451 5.76c0.973 0.012 1.76 0.798 1.771 1.77v7.755l10.043-9.085c0.127-0.102 0.272-0.191 0.427-0.261l0.013-0.005c0.217-0.105 0.473-0.166 0.742-0.166 0.001 0 0.002 0 0.003 0h-0c0.492 0.012 0.934 0.214 1.258 0.536l-0-0 0.002-0.002c0.037 0.033 0.071 0.067 0.103 0.102l0.001 0.001c0.255 0.305 0.41 0.701 0.41 1.134 0 0.346-0.099 0.669-0.271 0.942l0.004-0.007c-0.053 0.076-0.107 0.143-0.166 0.205l0.001-0.001c-0.068 0.080-0.141 0.151-0.221 0.215l-0.003 0.002-7.656 6.915 1.961 1.947 1.782-1.605c0.314-0.277 0.729-0.446 1.183-0.446 0.521 0 0.989 0.222 1.317 0.576l0.001 0.001c0.273 0.296 0.44 0.693 0.44 1.129 0 0.052-0.002 0.103-0.007 0.153l0-0.006c-0.029 0.48-0.248 0.904-0.583 1.201l-0.002 0.002-1.635 1.472 3.283 3.26 0.069 0.1 0.036 0.036c0.279 0.299 0.45 0.701 0.45 1.143 0 0.026-0.001 0.052-0.002 0.077l0-0.004c-0.012 0.472-0.21 0.895-0.524 1.201l-0 0c-0.284 0.291-0.671 0.48-1.102 0.508l-0.005 0c-0.041 0.003-0.089 0.005-0.137 0.005-0.37 0-0.713-0.112-0.998-0.305l0.006 0.004-0.061-0.048-0.028-0.026c-0.032-0.022-0.010-0.007-0.066-0.047-0.028-0.022-0.055-0.047-0.102-0.092l-3.453-3.443-1.782 1.605c-0.309 0.284-0.722 0.458-1.176 0.458-0.028 0-0.057-0.001-0.085-0.002l0.004 0c-0.487-0.008-0.922-0.224-1.222-0.562l-0.002-0.002c-0.277-0.314-0.446-0.729-0.446-1.184 0-0.52 0.222-0.989 0.576-1.316l0.001-0.001 1.64-1.477-1.952-1.946-2.070 1.876v4.73c0 0.003 0 0.006 0 0.010 0 0.974-0.789 1.763-1.763 1.763-0.491 0-0.935-0.201-1.255-0.524l-0-0c-0.324-0.313-0.525-0.752-0.525-1.237 0-0.004 0-0.008 0-0.011v0.001-17.254c0.012-0.973 0.798-1.76 1.77-1.771h0.001z"
          style="fill: #939393;"
        />
      </svg>
    </label>
    <span>
      1234567
    </span>
  </div>
</DocumentFragment>
`;
